pandas/core/arrays/sparse/accessor.py

```diff
@@ -227,15 +227,24 @@ def from_spmatrix(cls, data, index=None, columns=None):
         1  0.0  1.0  0.0
         2  0.0  0.0  1.0
         """
-        from pandas import DataFrame
+        from pandas import DataFrame, SparseDtype
+        from . import IntIndex, SparseArray
 
         data = data.tocsc()
         index, columns = cls._prep_index(data, index, columns)
-        sparrays = [SparseArray.from_spmatrix(data[:, i]) for i in range(data.shape[1])]
-        data = dict(enumerate(sparrays))
-        result = DataFrame(data, index=index)
-        result.columns = columns
-        return result
+        n_rows, n_columns = data.shape
+        data.sort_indices()
```

Comment:
It might be already done in `tocsc`, but that's a scipy implementation detail, and it doesn't really cost much. We need to make sure indices are sorted, since we create `IntIndex` with `check_integrity=False` that used to do this check.

Reply:
Maybe add a comment about that inline?